storage_service/common/utils.py

```diff
@@ -291,6 +306,253 @@ def get_compression(pointer_path):
+        try:
+            version = [
+                line
+                for line in subprocess.check_output("7z").splitlines()
+                if "Version" in line
+            ][0]
+            event_detail = 'program="7z"; version="{}"'.format(version)
+        except (subprocess.CalledProcessError, Exception):
+            event_detail = 'program="7z"'
+    elif compression in (COMPRESSION_TAR_BZIP2, COMPRESSION_TAR, COMPRESSION_TAR_GZIP):
+        try:
+            version = subprocess.check_output(["tar", "--version"]).splitlines()[0]
+            event_detail = 'program="tar"; version="{}"'.format(version)
+        except (subprocess.CalledProcessError, Exception):
+            event_detail = 'program="tar"'
+    else:
+        LOGGER.warning(
+            "Unknown compression algorithm, cannot correctly update pointer file"
+        )
+        event_detail = _("Unknown compression")
+
+    return event_detail
+
+
+def get_compression_transforms(compression, transform_order):
+    """Get descriptions of compression transforms performed on package
+
+    :param compression: one of the constants in ``COMPRESSION_ALGORITHMS``.
+    :param int transform_order: Starting value of the transform order
+    :return: list of `etree.Element`s
+    """
+    transforms = []
+    if compression in (COMPRESSION_7Z_BZIP, COMPRESSION_7Z_LZMA, COMPRESSION_7Z_COPY):
+        if compression == COMPRESSION_7Z_BZIP:
+            algo = "bzip2"
+        elif compression == COMPRESSION_7Z_LZMA:
+            algo = "lzma"
+        elif compression == COMPRESSION_7Z_COPY:
+            algo = "copy"
+        transforms.append(
+            etree.Element(
+                PREFIX_NS["mets"] + "transformFile",
+                TRANSFORMORDER=str(transform_order),
+                TRANSFORMTYPE="decompression",
+                TRANSFORMALGORITHM=algo,
+            )
+        )
+
+    elif compression in (COMPRESSION_TAR_BZIP2, COMPRESSION_TAR):
+        if compression == COMPRESSION_TAR_BZIP2:
+            transforms.append(
+                etree.Element(
+                    PREFIX_NS["mets"] + "transformFile",
+                    TRANSFORMORDER=str(transform_order),
+                    TRANSFORMTYPE="decompression",
+                    TRANSFORMALGORITHM="bzip2",
+                )
+            )
+            transform_order += 1
+
+        transforms.append(
+            etree.Element(
+                PREFIX_NS["mets"] + "transformFile",
+                TRANSFORMORDER=str(transform_order),
+                TRANSFORMTYPE="decompression",
+                TRANSFORMALGORITHM="tar",
+            )
+        )
+
+    elif compression == COMPRESSION_TAR_GZIP:
+        transforms.append(
+            etree.Element(
+                PREFIX_NS["mets"] + "transformFile",
+                TRANSFORMORDER=str(transform_order),
+                TRANSFORMTYPE="decompression",
+                TRANSFORMALGORITHM="gzip",
```

Comment:
I don't know whether we have a preference. Currently the code adds a separate compression transform element for `tar`. Same as `pbzip2` compression method. But this can change. I developed the gzip additions using the pbzip2 compression as a template (since they both use tar and seem to be very similar) - whatever is decided about this, should `pbzip2` be treated the same way?

Reply:
Thanks @helenst "tar gzip" would be great in the `TRANSFORMALGORITHM` too. It wouldn't normally be your responsibility making a PR like this to change `pbzip2` at the same time, but if you had a spare moment. Only annoying thing is that the change needs to be made in Archivematica as far as I can tell.

I had to investigate this today as I noticed pbzip2 is outputting blank: 

```
      <mets:file ID="pbzip-ae0b9327-1b7f-43f9-8e07-4be4830259c5.tar" GROUPID="Group-ae0b9327-1b7f-43f9-8e07-4be4830259c5" ADMID="amdSec_113644">
        <mets:FLocat xlink:href="/var/archivematica/sharedDirectory/www/AIPsStore/ae0b/9327/1b7f/43f9/8e07/4be4/8302/59c5/pbzip-ae0b9327-1b7f-43f9-8e07-4be4830259c5.tar.bz2" LOCTYPE="OTHER" OTHERLOCTYPE="SYSTEM"/>
        <mets:transformFile TRANSFORMTYPE="decompression" TRANSFORMORDER="1" TRANSFORMALGORITHM=""/>
      </mets:file>
```

It turns out there needs to be a second component in `workflow.json`, 

* [pbzip2](https://github.com/artefactual/archivematica/blob/qa/1.x/src/MCPServer/lib/assets/workflow.json#L1056)
* [gzip workflow](https://github.com/artefactual/archivematica/blob/qa/1.x/src/MCPServer/lib/assets/workflow.json#L1044)

Would need to read something like:

* `"pbzip-tar.pbzip2"`,
* `"gzip-tar.gzip":`

So that here: https://github.com/artefactual/archivematica/blob/qa/1.x/src/MCPClient/lib/clientScripts/compress_aip.py#L45 the algorithm is picked up. 

That will automatically be picked up for `pbzip2` but then for gzip, we need to add a string replacement `{}` [here](https://github.com/artefactual/archivematica/blob/9a6fe32f2dc847038acf895a4356d0dbc8e7779d/src/MCPClient/lib/clientScripts/compress_aip.py#L105-L106) to match the other functions. 

I should have picked up on this before.